src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,175 @@
+        assertEquals(actual, john);
+        User nextUserToAdd = registrationService.register(maria);
+        assertEquals(nextUserToAdd, maria);
+    }
+
+    @Test
+    void register_underageUser_NotOk() {
+        john.setAge(TEEN_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_inadmissibleAge_NotOK() {
+        george.setAge(ZERO_AGE);
+        george.setLogin(anotherValidLogin);
+        george.setPassword(anotherValidPassword);
+        jane.setAge(NEGATIVE_AGE);
+        jane.setLogin(thirdValidLogin);
+        jane.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(jane));
+    }
+
+    @Test
+    void register_shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        george.setAge(ANOTHER_ADULT_AGE);
+        george.setLogin(shortLogin);
+        george.setPassword(anotherValidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_shortPassword_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(invalidPassword);
+        george.setAge(ADULT_AGE);
+        george.setLogin(thirdValidLogin);
+        george.setPassword(shortPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+    }
+
+    @Test
+    void register_emptyLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin("");
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_emptyPassword_NotOK() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(null);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_nullPassword_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void register_duplicateUser_NotOk() {
```

Comment:
The test `register_duplicateUser_NotOk` is using `Storage.people.add()` directly which is correct as per the checklist. However, ensure you have a separate test for validating that a successful registration adds the user to the storage.